pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -506,6 +532,16 @@ func (r *Reconciler) applyNewResources(ctx context.Context, log logr.Logger, ori
 			if err := injectLabels(obj.obj, labelsToInject); err != nil {
 				return fmt.Errorf("error injecting labels into object %q: %s", resource, err)
 			}
+			// ref https://kubernetes.io/docs/reference/labels-annotations-taints/#kubectl-k8s-io-restart-at
+			if err := preserveAnnotation("Deployment", "apps/v1", current, obj.obj, "kubectl.kubernetes.io/restartedAt", "spec", "template", "metadata", "annotations"); err != nil {
+				return fmt.Errorf("could not preserve annotation : %w", err)
+			}
+			if err := preserveAnnotation("DaemonSet", "apps/v1", current, obj.obj, "kubectl.kubernetes.io/restartedAt", "spec", "template", "metadata", "annotations"); err != nil {
+				return fmt.Errorf("could not preserve annotation : %w", err)
+			}
+			if err := preserveAnnotation("StatefulSet", "apps/v1", current, obj.obj, "kubectl.kubernetes.io/restartedAt", "spec", "template", "metadata", "annotations"); err != nil {
+				return fmt.Errorf("could not preserve annotation : %w", err)
+			}
```

Comment:
Use switch-case here? I think the function doesn't need to be called three times in a row, nor does the `preserveAnnotation` requires kind and apiVersion.